tests/epyccel/test_epyccel_headers.py

```diff
@@ -0,0 +1,73 @@
+# pylint: disable=missing-function-docstring, missing-module-docstring, missing-class-docstring
+# coding: utf-8
+""" Tests for headers. This ensures intermediate steps are tested before headers are deprecated.
+Once headers are deprecated this file can be removed.
+"""
+
+import warnings
+import pytest
+
+from pyccel.epyccel import epyccel
+from pyccel.errors.errors import PyccelSemanticError
+from pyccel.decorators import allow_negative_index, stack_array
+
+
+def test_local_type_annotation(language):
+    def local_type_annotation():
+        #$ header variable gift int
+        gift = 10
+        return gift
+
+    epyc_local_type_annotation = epyccel(local_type_annotation, language=language)
+    assert epyc_local_type_annotation() == local_type_annotation()
+    assert isinstance(epyc_local_type_annotation(), type(local_type_annotation()))
+
+def test_local_wrong_type_annotation(language):
+    def local_wrong_type_annotation():
+        #$ header variable gift float
+        gift = 10
+        return gift
+
+    with pytest.raises(PyccelSemanticError):
+        epyccel(local_wrong_type_annotation, language=language)
+
+def test_allow_negative_index_annotation(language):
+    @allow_negative_index('array')
+    def allow_negative_index_annotation():
+        import numpy as np
+        #$ header variable array int[:](order=C)
+        array = np.array([1,2,3,4,5])
+        j = -3
+        return array[j]
+
+    with warnings.catch_warnings(record=True) as record1:
+        warnings.simplefilter("always")
+        epyc_allow_negative_index_annotation = epyccel(allow_negative_index_annotation, language=language)
+    assert len(record1) == 1
+
+    assert epyc_allow_negative_index_annotation() == allow_negative_index_annotation()
+    assert isinstance(epyc_allow_negative_index_annotation(), type(allow_negative_index_annotation()))
+
+
+def test_stack_array_annotation(language):
+    @stack_array('array')
+    def stack_array_annotation():
+        import numpy as np
+        #$ header variable array int[:,:]
+        array = np.array([[1,2],[3,4],[5,6]])
+        return array[2,0]
+
+    with pytest.raises(PyccelSemanticError):
+        epyccel(stack_array_annotation, language=language)
```

Comment:
Does this mean that a stack array cannot be declared through a header? Or do we get a similar error using a variable annotation?

Reply:
Yes this is correct. This is because the array is created (including its fixed size) at line 56. We don't have information about its shape here so this is not possible. In one of the next few PRs I will fix #1528 . This will allow us to translate:
```
array : 'int[:,:]' = np.array([[1,2],[3,4],[5,6]])
```
Here all the information is in the same line. So there is no issue.

That said, the usual usage for this is in a header file where the declaration won't be printed. We may want to think about extending the syntax to allow a declaration of:
```
#$ header variable array int[2,3]
```

Reply:
I see, thanks for the explanation!

The header file (or stub file) will declare the type of an array variable including its rank, but I do not see the need to know its exact shape given that we do not provide it for function arguments.

Reply:
You are right. Fixed def16c2b